chore/ci-checkstyle-javadoc.sh

```diff
@@ -37,41 +42,70 @@ function create_checkstyle_config() {
 </module>' > "$JAVADOC_CHECKSTYLE_CONFIG"
 }
 
+function run_checkstyle() {
+    create_checkstyle_config
+    mvn -B checkstyle:check --fail-never -Dcheckstyle.config.location="$JAVADOC_CHECKSTYLE_CONFIG"
+}
+
 function compute_num_errors() {
-    echo $(mvn -B checkstyle:check --fail-never -Dcheckstyle.config.location="$JAVADOC_CHECKSTYLE_CONFIG" \
-        | grep -Po '(?<=There are )\d+(?= errors reported by Checkstyle)')
+     grep -Po '(?<=There are )\d+(?= errors reported by Checkstyle)' <<< `run_checkstyle`
 }
 
 function main() {
     cd "$(git rev-parse --show-toplevel)"
 
     # compute compare score
     git checkout --force "$COMPARE_BRANCH" &> /dev/null
-    create_checkstyle_config
     compare_num_errors=`compute_num_errors`
 
     # compute current score
     git checkout --force - &> /dev/null
-    create_checkstyle_config
     current_num_errors=`compute_num_errors`
 
     echo "JAVADOC QUALITY SCORE (lower is better)
     Compare: $compare_num_errors
     Current: $current_num_errors
     "
 
-    if [[ -z $compare_num_errors ]]; then
+    if [ -z $compare_num_errors ]; then
         echo "Failed to compute compare score";
         exit 1;
-    elif [[ -z $current_num_errors ]]; then
+    elif [ -z $current_num_errors ]; then
         echo "Failed to compute current score";
         exit 1;
-    elif [[ $compare_num_errors < $current_num_errors ]]; then
+    elif [ $compare_num_errors -lt $current_num_errors ]; then
```

Comment:
Technically, this is a bugfix masquerading as a compatibility improvement, as `<` in extended expression syntax actually does a lexicographical comparison. That is to say, something like `[[ 10000 < 2 ]]` evaluates to true, whereas `-lt` actually does a numeric less-than comparison. Oopsie!